### DISCUSSION TITLE
I can access `<MyPublicIP>:80` but not `<MyPublicIP>:443` from local network but works fine when accessed from external network or mobile data

### DISCUSSION THREAD

**Discussion by dinbtechit** [NginxProxyManager/nginx-proxy-manager#3641] [Q&A]
I am running into a weird issue with Nginx Proxy Manager. This might not be an issue with NPM probably related to router. But I'm unable to access applications via HTTPS in Local network however works fine from my mobile data or external network . (HTTP Only services work fine). Has anyone else face similar issues?

**My Setup:**
1. Home Router - Port Forwarded 80 and 443 (I can access `<MyPublicIP>:80` but not `<MyPublicIP>:443`)
2. Using - Dynamic DNS
3. Nginx Proxy manager -> toward the  application running locally.

**Comment by AhilusUA** [comment]
Mikrotik?

**Comment by forestial** [comment]
I have just posted another new discussion thread (Aug 7) with what sounds like the same issue.  (Apologies for the duplication; I did search but I didn't find this thread until now).

OP, Did you find a solution to this?

**Reply by AhilusUA** [reply]
```
add address=192.168.1.0/24 list=vlan1
add address=192.168.1.x list=nginx
add address=x.x.x.x list=public_ip
add action=masquerade chain=srcnat comment="local HTTP/S to nginx" \
    dst-address-list=nginx dst-port=80,443 protocol=tcp src-address-list=\
    vlan1
add action=dst-nat chain=dstnat dst-address-list=public_ip dst-port=80,443 \
    protocol=tcp src-address-list=vlan1 to-addresses=*nginx_ip*
```
``nginx`` address list is internal IP for nginx-proxy (192.168.1.x)
``vlan1`` address list is internal subnet (192.168.1.0/24) 
``public_ip`` address list for my external IP
``nginx_ip`` it's just internal IP for nginx (192.168.1.x)

you can read about hairpin nat, it's about access to internal resources through an external IP from inside the network

**Reply by forestial** [reply]
Thanks.  I will review Hairpin NAT.

I found after posting my question that my problem appears to be a limitation of my ISP (Optimum/Altice), or the router that they provide and how they handle port 443 for internal clients.  I was able to work around it (while still using NPM) by getting my HTTPS request to use a different port.  It's a small inconvenience to add the port number to any request, and it seems to work for both internal and external access.
